### PR TITLE
set the correct vars in comp vector

### PR DESCRIPTION
**Pull request by rfomin**
I made the mistake of not resetting the `comp` vector for `complevel vanilla`.

This requires additional testing, not for merge.

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
Hm, shouldn't `comp[i]` always have value `1` for Vanilla and always have the value of `compatibility` for Boom?

**Comment by rfomin** [comment]
> shouldn't `comp[i]` always have value `1` for Vanilla

Yes, but I forgot to set it for demo recording / gameplay.


> always have the value of `compatibility` for Boom

No, `compatibility` is always 0. At least for this: 
```
    // [FG] In Boom, monsters did not stay on a lift
    comp[comp_staylift] = 1;
    // [FG] Boom did not prevent zombies from exiting levels
    comp[comp_zombie] = 1;
```
and PrBoom set more vars than that.

**Comment by fabiangreffrath** [comment]
This is slowly approaching the level of complexity that gives me the shivers in PrBoom's code...

**Comment by rfomin** [comment]
Well, we could not add the compatibility matrix, just set `comp[i] = 1` for the `vanilla complevel`. Lee Killough added
`&& demo_version >= 203` in most places anyway, with a few exceptions.

But I think this is more readable, than `memset(comp, 0xff, sizeof comp);`

**Comment by fabiangreffrath** [comment]
> Well, we could not add the compatibility matrix, just set `comp[i] = 1` for the `vanilla complevel`. Lee Killough added
> `&& demo_version >= 203` in most places anyway, with a few exceptions.

Yes, please. Let's just set the whole `comp[]` array to `1` and add `&& demo_version >= 203` where Lee forgot it.

> But I think this is more readable, than `memset(comp, 0xff, sizeof comp);`

Yes, but that's original MBF code. 😉

**Comment by rfomin** [comment]
Maybe we shouldn't draw the "Doom Compatibility" menu for the `complevel < 203`? I think in that case this menu is just confusing.

File: Source/g_game.c [hunk 1]
```diff
@@ -2789,14 +2806,7 @@ byte *G_ReadOptions(byte *demo_p)
     }
   else  // defaults for versions < 2.02
     {
-      int i;  // killough 10/98: a compatibility vector now
-      for (i=0; i < COMP_TOTAL; i++)
-	comp[i] = compatibility;
```
Comment (fabiangreffrath) [inline Source/g_game.c:2794]: So, we do not consider Boom's compatibility flag anymore?

Comment (rfomin) [reply Source/g_game.c:2794]: It's always zero in PrBoom+ demos:
https://github.com/coelckers/prboom-plus/blob/57750b80f9787f2fdb1b87a0dc84588fb66ab6da/prboom2/src/g_game.c#L3483-L3484

Comment (fabiangreffrath) [reply Source/g_game.c:2794]: But one line above there is `boom_compatibility_compatibility`.

Comment (rfomin) [reply Source/g_game.c:2794]: But we do not support `boom_compatibility_compatibility`. I've never seen them, but I guess we should keep `compatibility` for demo playback.

File: Source/g_game.c [hunk 2]
```diff
@@ -2345,6 +2345,17 @@ static int G_GetDefaultComplevel()
   }
 }
 
+static void G_BoomComp()
```
Comment (fabiangreffrath) [inline Source/g_game.c:2348]: Strictly speaking, we don't even need to factor this out anymore.

**Review by fabiangreffrath — APPROVED** [review]
LGTM, is this ready for merging?

**Comment by rfomin** [comment]
Wait a bit, toxic.rat from dw has  `complevel vanilla`  desync and I'm trying to fix it in private messages.

**Comment by fabiangreffrath** [comment]
Alright, please keep me updated.

**Comment by rfomin** [comment]
I did some vanilla demo tests and there is deync because we are not doing REJECT overflow emulation. Is this a deliberate choice? 00e17c0e1fe95a922957ba5e458c2ab514cd3650

**Comment by fabiangreffrath** [comment]
This was some "at least it doesn't crash anymore" kind of choice. Demo playback wasn't that much of an issue back then. Sure, it should get fixed.

**Comment by fabiangreffrath** [comment]
Are this one and the REJECT overflow emulation ready for merging now?

**Comment by rfomin** [comment]
Yes, I think so. I didn’t get a response from toxic.rat, but everything works according to my tests.

**Pull request merged**